sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -151,23 +151,96 @@ AZ_NODISCARD az_result az_iot_provisioning_client_register_get_subscribe_topic_f
 
   az_span mqtt_topic_filter_span
       = az_span_init((uint8_t*)mqtt_topic_filter, (int32_t)mqtt_topic_filter_size);
-  int32_t required_length = 1 + az_span_size(str_dps_registrations_res);
+  int32_t required_size = 1 + 1 + az_span_size(str_dps_registrations_res);
```

Comment:
can we make an explicit call to `sizeof()`?
```suggestion
  int32_t required_size = (int32_t)sizeof('#') + (int32_t)sizeof('\0') + az_span_size(str_dps_registrations_res);
```

Reply:
But in this case, along with my below suggestion, the sizeof null would be moved to the `AZ_RETURN_IF_NOT_ENOUGH_SIZE`

Reply:
That would be incorrect as '#' is an `int`. So in C, sizeof('#') would be 4. Which means your code would always require 6 bytes extra.
It's also extremely confusing as it will work as expected in C++ where sizeof('#') is 1.

Please check other code if you believe we have that mistake.